src/bp-core/admin/bp-core-admin-tools.php

```diff
@@ -1126,7 +1135,84 @@ function bp_admin_repair_tools_wrapper_function() {
 		$status = bp_admin_reinstall_emails();
 	} elseif ( 'bp-assign-member-type' === $type ) {
 		$status = bp_admin_assign_member_type();
+	} elseif ( 'bp-sync-activity-favourite' === $type ) {
+		$status = bp_admin_update_activity_favourite();
 	}
 	wp_send_json_success( $status );
 }
 add_action( 'wp_ajax_bp_admin_repair_tools_wrapper_function', 'bp_admin_repair_tools_wrapper_function' );
+
+/**
+ * Check if BuddyPress activity favorites data needs upgrade & Update to BuddyBoss activity like data
+ *
+ * @since BuddyBoss 1.3.2
+ */
+function bp_admin_update_activity_favourite() {
+
+	$bp_activity_favorites = bp_get_option( 'bp_activity_favorites', false );
+
+	if ( ! $bp_activity_favorites ) {
+
+		$offset = isset( $_POST['offset'] ) ? (int) ( $_POST['offset'] ) : 0;
+
+		$args = array(
+			'number' => 50,
+			'offset' => $offset,
+		);
+
+		$users = get_users( $args );
+
+		if ( ! empty( $users ) ) {
+
+		    foreach ( $users as $user ) {
+			    $my_favs = bp_get_user_meta( $user->ID, 'bp_favorite_activities', true );
```

Comment:
@chetansatasiya change the name of variable to `$user_favs` here and other places as well.

Reply:
@mehulkaklotar made the changes.